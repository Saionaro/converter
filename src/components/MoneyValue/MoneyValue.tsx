import React from "react";
import { Currency } from "src/common-types";
import { formatters } from "src/utils/format";

interface Props {
  currency: Currency;
  children: string;
}

export function MoneyValue({ children, currency }: Props) {
  return <span>{formatters[currency](children)}</span>;
}
