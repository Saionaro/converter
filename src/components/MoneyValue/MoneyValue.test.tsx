import React from "react";
import { render } from "@testing-library/react";
import { MoneyValue } from "./MoneyValue";

import "@testing-library/jest-dom/extend-expect";

describe("components/MoneyValue", () => {
  test("renders a formatted money", () => {
    const { container, getByText } = render(
      <MoneyValue currency="USD">5400</MoneyValue>
    );
    expect(getByText("$5,400.00")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders a correct zero", () => {
    const { container, getByText } = render(
      <MoneyValue currency="USD">0</MoneyValue>
    );
    expect(getByText("$0.00")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
