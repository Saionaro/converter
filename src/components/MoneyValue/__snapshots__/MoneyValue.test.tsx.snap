// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MoneyValue renders a correct zero 1`] = `
<span>
  $0.00
</span>
`;

exports[`components/MoneyValue renders a formatted money 1`] = `
<span>
  $5,400.00
</span>
`;
